### ISSUE TITLE
Build tutorial request

### ISSUE THREAD

**Issue by Muph0** [aljen/spaghetti#51]
Hello, I know this is probably an inappropriate place to ask,
but I'm struggling to build this project with visual studio cmake.

Could you maybe include a small tutorial how to do it from start to finish, @aljen?
That would be great so I could finally tinker with your marvellous creation.

**Comment by aljen** [comment]
Hi, it's the best place to ask, that's what the issues are for.
Sorry for long time in response, I was on vacation :smiley:

Sure, I'll try to do this on the weekend

**Comment by ramadom908** [comment]
Hello Aljen, are you still able to do a tutorial about using cmake to build the project. I also am having problems with cmake on windows. Are there any other alternatives to cmake?

**Comment by aljen** [comment]
Hello, I'm sorry, I'm still a little busy at work, that's why it's been going on for so long.

I'm using Qt Creator on Linux and on Windows as an IDE.

- Install **Visual Studio 2017**, **CMake** & **Qt5** (make sure to select **MSVC 2017 64-bit** in **Qt** -> **Qt 5.11.2** and to select **Qt Charts**)

- Configure **MSVC** as a compiler in **Qt Creator**

- And just open **CMakeLists.txt** in the root directory of the repo (select only **Debug** & **Release**)

**Comment by ramadom908** [comment]
OK, so I build the project but when I run it, I get SpaghettiEditor.exe has stoped working... and in QT editor  SpaghettiEditor.exe has exited with code 3.

I did run windeployqt app from QT bin and also searched for some other dlls like msvcp140.dll and added them to the bin folder of my project... but still i get the same result the app simply closes. Am i missing something? I can't eve run in from QT editor....

**Comment by aljen** [comment]
My bad :D
In build directory like ```build-spaghetti-Desktop_Qt_5_11_2_MSVC2017_64bit-Debug``` all you need to do is create **Packages** directory, that should do it :)

**Comment by ramadom908** [comment]
And make an installer? Shouldn't i be able to open the spaghettieditor  with QTCreator?  Should I get this librarys installed on my pc?   https://github.com/nlohmann/json , https://github.com/cameron314/concurrentqueue, https://github.com/greg7mdp/sparsepp,    http://www.boost.org  .

I run  the debuger and i got all sorts of errors regarding  json.hpp

**Comment by aljen** [comment]
Nope, everything needed is already in the tree, except spdlog, which is pulled as a git submodule.
On the release page there are a few old releases, packed as .7z, so installer is not an option for now - I simply don't have time to make one.

**Comment by ramadom908** [comment]
Come on really????? all i had to do was put a simple empty folder called "packages" in the release folder??? That worked, i didn't really understood you the first the since it was a simple thing....(now it's obvious)  I was strugling for 2 days with this.... good thing I figured it out

**Comment by ramadom908** [comment]
Thank you for your answers Aljen!

**Comment by aljen** [comment]
Yeah, that simple :)
In the hindsight I should check if it exists before using it like every other directory, I'll try to add it in the weekend :)

Np, thanks for valuable input and for the interest in this project! :)

**Comment by ramadom908** [comment]
Hi Aljen, from what I can see there is the ability to pack certain circuits in a package, is this fully implemented and I don't know how to use it? or it's not yet working? If it is how can I make it work ? 

Never mind found it :D bottom left under packages (for anyone wondering the same thing). Just have to input the path close editor and open again.

**Comment by aljen** [comment]
Hey,

yes it's fully implemented, all you need to do is to set a path in a package properties, save it and after restart it should be available in the **Packages** section on the left panel :)

There is no live reload at the moment, so you must restart it to see it in **Packages** section :/

Glad you found it by yourself :)

**Comment by Hercules2905** [comment]
Hi aljen,

Any updates on the tutorials? I'm completely new to the process of building C++ files. I have git installed, but CMake cannot find the necessary spdlog files. A tutorial would be really nice.

Thank you!
